### ISSUE TITLE
We can't access local variables inside closures?

### ISSUE THREAD

**Issue by raj-rajaratnam** [ballerina-platform/ballerina-lang#15068]
**Description:**

This works
```
import ballerina/io;

public type Order record {
    int id;
    string name;
};

table<Order> orderTable = table {
    { id, name },
    [
        {1, "BTC"},
        {2, "LTC"}
    ]
};

public function main() {
    io:println(orderTable);
    var s = orderTable.remove(function (Order ord) returns boolean {
                        return ord.id > 0;
                    });
    io:println(orderTable);
}
```

But when I use a local variable inside closure, it doesn't work (it doesn't remove the rows from the table)

```
    io:println(orderTable);
    int invalid = 0;
    var s = orderTable.remove(function (Order ord) returns boolean {
                        return ord.id > invalid;
                    });
    io:println(orderTable);
```

But when I move that variable to a global scope, it works. So is this a bug that i'm not able to use local variable inside closure?

**Comment by raj-rajaratnam** [comment]
@hasithaa @anupama-pathirage can you pls prioritize this?

**Comment by pubudu91** [comment]
Duplicate of #15047. This was fixed with PR #15059.

**Comment by pubudu91** [comment]
Closing since this was fixed with above mentioned PR.
